engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -62,6 +66,7 @@
 
   private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(LoggerUtil.class);
 
+  private static boolean setMDCProperties = true;
```

Comment:
It should be called only once and there is no need to call the function for each appender. In any case, MDC information is static and calling the method to populate MDC multiple times does not affect anything (MDC will be populated with the same information).